doc/Type/Mu.pod6

```diff
@@ -343,86 +343,9 @@ case we are declaring this C<new> method simply to avoid the extra syntax of
 using pairs when creating the object. C<self.bless> returns the object, which is
 in turn returned by C<new>.
 
-However, in general, implementing a customized C<new> method might not be the best
-way of initializing a class, even more so if the default constructor is
-disabled, since it can make it harder to correctly initialize the class from a
-subclass. For instance, in the above example, the C<new> implementation takes
-two positional arguments that must be passed from the subclass to the superclass
-in the exact order. That is not a real problem if it's documented, but take
-into account C<bless> will eventually be calling C<BUILD>
-in the class that is being instantiated. This might result in some unwanted
-problems, like having to create a C<BUILD> submethod to serve it correctly:
-
-=begin code
-class Point {
-    has Int $.x;
-    has Int $.y;
-    multi method new($x, $y) {
-        self.bless(:$x, :$y);
-    }
-}
-
-class Point-with-ID is Point {
-    has Int $.ID  is rw = 0;
-
-    submethod BUILD( *%args ) {
-        say %args;                # OUTPUT: «{x => 1, y => 2}␤»
-        for self.^attributes -> $attr {
-            if $attr.Str ~~ /ID/ {
-                $attr.set_value( self, "*" ~ %args<x> ~ "-" ~ %args<y> ) ;
-            }
-        }
-    }
-}
-
-my $p = Point-with-ID.new(1,2);
-say $p.raku;
-# OUTPUT: «Point-with-ID.new(ID => "*1-2", x => 1, y => 2)␤»
-=end code
-
```

Comment:
is there a point in deleting all this example? Is there something somewhere else that exemplifies the use of *%args? Or introspection?

Reply:
The introductory text for this example states that the complex way of setting the `$!ID` attribute is necessary when implementing a custom `new()`. The following code seems to behave the same and has none of the complexity though:

```
class Point-with-ID is Point {
    has Str $.ID  is rw = '0';

    submethod BUILD( :$x, :$y ) {
        $!ID = "*" ~ $x ~ "-" ~ $y;
    }
}
```

If the above code is actually equivalent, then the text and example are wrong in that the complexity is in any way necessary. That's the reason why I deleted them. They state and suggest a wrong fact. (Please correct me if I just missed the point of the example and there actually is a reason for doing it this way!)

I see your point about not removing parts of the documentation that provide insight that is not better explained elsewhere. I searched a bit and found the following.

There is a bit about `.^attributes` in <https://docs.raku.org/language/classtut#Introspection>.
The `*%args` slurpy hash is explained in <https://docs.raku.org/type/Signature#Slurpy_(A.K.A._variadic)_parameters>.
Maybe we can add the interesting bits of this example there?

Reply:
Well, main point was and is and deleting something needs a bit of discussion. I take your point, but let me do a bit of research to find the reason why that was written that way.

Reply:
The explanation for this code is in #2077. It's related to `bless`, the whole point being that that `BUILD` is going to be called from the superclass' `new`. Your code is probably a better choice. But in that case, the best way would have been to reopen #2077 and link this change to that issue.

Reply:
I have reopened #2077 and changed this PR to close that ticket as you suggested.